bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -463,6 +469,51 @@ private static boolean isConvertibleToArrayType(Object sourceValue, BArrayType t
         return true;
     }
 
+    private static boolean isConvertibleToTupleType(Object sourceValue, BTupleType targetType,
+                                                    List<TypeValuePair> unresolvedValues) {
+        if (!(sourceValue instanceof ArrayValue)) {
+            return false;
+        }
+
+        ArrayValue source = (ArrayValue) sourceValue;
+        List<Type> targetTypes = targetType.getTupleTypes();
+        int sourceTypeSize = source.size();
+        int targetTypeSize = targetTypes.size();
+        Type targetRestType = targetType.getRestType();
+
+        if (sourceTypeSize < targetTypeSize) {
+            return false;
+        }
+        if (targetRestType == null && sourceTypeSize > targetTypeSize) {
+            return false;
+        }
+
+        Set<Type> convertibleTypes;
+
+        for (int i = 0; i < targetTypeSize; i++) {
+            convertibleTypes = getConvertibleTypes(source.getRefValue(i), targetTypes.get(i), unresolvedValues);
+            if (convertibleTypes.isEmpty()) {
+                return false;
+            }
+            if (convertibleTypes.size() > 1 && !convertibleTypes.contains(TypeChecker.getType(source.getRefValue(i)))
+                    && !hasIntegerSubTypes(convertibleTypes)) {
+                return false;
+            }
+        }
```

Comment:
Shall we move this logic to a separate method?
Since we the only difference is the `targetTypes.get(i)` part, we can pass it as a parameter

Reply:
Following part has been moved to a separate method where we can reuse the code for the method `isConvertibleToArrayType` also. 
```
convertibleTypes = getConvertibleTypes(source.getRefValue(i), targetTypes.get(i), unresolvedValues);
if (convertibleTypes.isEmpty()) {
    return false;
}
if (convertibleTypes.size() > 1 && !convertibleTypes.contains(TypeChecker.getType(source.getRefValue(i)))
        && !hasIntegerSubTypes(convertibleTypes)) {
    return false;
}
```